src/main/java/io/prometheus/cloudwatch/CachingDimensionSource.java

```diff
@@ -0,0 +1,132 @@
+package io.prometheus.cloudwatch;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+final class CachingDimensionSource implements DimensionSource {
+
+  private static final int MAX_ENTRIES = 500;
```

Comment:
How did you choose this number? Under what circumstances would someone want to use a different number?